### PR TITLE
allow setting transaction isolation

### PR DESCRIPTION
**Pull request by bbeaudreault**
@sougou this is as we discussed -- vttablet enforces by executing `set transaction isolation level` before each `Begin`, when value is non-default.  See https://dev.mysql.com/doc/refman/5.7/en/set-transaction.html

I think this is all it takes, let me know if you think I missed something.

### CODE REVIEW

File: go/vt/vttablet/tabletserver/tx_engine_test.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 
 	"github.com/youtube/vitess/go/vt/vttablet/tabletserver/tabletenv"
 
+	"github.com/youtube/vitess/go/vt/proto/query"
```
Comment (sougou) [inline go/vt/vttablet/tabletserver/tx_engine_test.go]: This should be renamed to `querypb "github...`.

I have a patch you can apply to goimports to do this automatically: https://gist.github.com/sougou/2255bc77bf8eaa45eadcc44f8832514f

File: go/vt/vttablet/tabletserver/tx_pool.go [hunk 2]
```diff
@@ -182,6 +182,20 @@ func (axp *TxPool) Begin(ctx context.Context, useFoundRows bool) (int64, error)
 		}
 		return 0, err
 	}
+
+	switch txIsolation {
+	case querypb.ExecuteOptions_DEFAULT:
+		break
+	case querypb.ExecuteOptions_REPEATABLE_READ, querypb.ExecuteOptions_READ_COMMITTED, querypb.ExecuteOptions_READ_UNCOMMITTED, querypb.ExecuteOptions_SERIALIZABLE:
+		query := fmt.Sprintf("set transaction isolation level %v", strings.Replace(txIsolation.String(), "_", " ", 1))
```
Comment (sougou) [inline go/vt/vttablet/tabletserver/tx_pool.go]: I think it's better to use a `map[querypb.ExecuteOptions]string` here. We could even map it to the full statement, thereby avoiding the Sprintf.

File: go/vt/vttablet/tabletserver/tx_pool_test.go [hunk 3]
```diff
@@ -31,6 +31,7 @@ import (
 	"github.com/youtube/vitess/go/vt/vterrors"
 	"github.com/youtube/vitess/go/vt/vttablet/tabletserver/tabletenv"
 
+	"github.com/youtube/vitess/go/vt/proto/query"
```
Comment (sougou) [inline go/vt/vttablet/tabletserver/tx_pool_test.go]: use `querypb`

**Comment by bbeaudreault** [comment]
@sougou fixed, PTAL

**Comment by bbeaudreault** [comment]
Not sure why `make proto` didn't generate the python the first time, but should be fixed now.

**Comment by sougou** [comment]
LGTM

**Pull request merged**